config/loader.go

```diff
@@ -0,0 +1,202 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/asaskevich/govalidator"
+	"github.com/mitchellh/mapstructure"
+	"github.com/spf13/viper"
+	"golang.org/x/text/cases"
+)
+
+var ErrNilConfig = errors.New("nil configuration")
+
+// IValidatable defines an interface of objects that can self-validate.
+type IValidatable interface {
+	Validate() error
+}
+
+// Loader is responsible for loading config from a source into a struct instance.
+type Loader struct {
+	config    *mapstructure.DecoderConfig
+	exclusive bool
+}
+
+// NewLoader creates a new loader. dest must be a pointer to a struct instance
+// to be populated, other wise, nil returned.
+func NewLoader(dest interface{}) *Loader {
+	loader := &Loader{exclusive: true}
+
+	if err := loader.Init(dest); err != nil {
+		panic(err)
```

Comment:
The only reason this would fail is if `dest` provided is not a struct pointer. This is a programming error that is only resolvable by removing the bug from the calling code. Panicking seems appropriate, and more robust then relying on the calling code to check the error.